client/src/components/editor/ArticleUpdateForm.js

```diff
@@ -22,13 +35,75 @@ class ArticleUpdateForm extends Component {
     }
   }
 
+  async componentDidUpdate(prevProps, prevState) {
+    // Cannot use onMount here because App needs a moment to async fetch web3 and the contract.
+    if (this.props.editorContract !== prevProps.editorContract) {
+        const articleHashPromise = this.props.editorContract?.methods?.getArticleHash(this.props.match.params.id).call().then((articleHash) => {
+          this.setState({
+            ipfsHash: window.web3.utils.hexToAscii(articleHash[articleHash.length - 1])
+          });
+        });
+        const articlesPromise = this.props.editorContract?.methods?.getArticles().call().then((articles) => {
```

Comment:
`this.props.editorContract?` is this some sort of shorthand for checking if the contract is defined?

Reply:
yep yep https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
Gotcha. I think it's unnecessary here, as if the editorContract has changed between prevProps, it's going to be defined.

Reply:
If its not, it's because you didn't include a componentDidMount and then you're forcing an update here.

Reply:
removed these